InMobi/src/main/java/com/mopub/nativeads/InMobiNative.kt

```diff
@@ -0,0 +1,320 @@
+    ) {
+        mNativeListener = customEventNativeListener
+
+        try {
+            mPlacementId = InMobiAdapterConfiguration.getPlacementId(serverExtras)
+        } catch (placementIdException: InMobiAdapterConfiguration.InMobiPlacementIdException) {
+            val errorMsg =
+                "InMobi Native request failed. Placement Id is not available or incorrect. Please make sure you set valid Placement Id on MoPub UI."
+            onError(customEventNativeListener, errorMsg)
+            return
+        }
+
+        InMobiAdapterConfiguration.initializeInMobi(
+            serverExtras,
+            context,
+            object : InMobiAdapterConfiguration.InMobiInitCompletionListener {
+                override fun onSuccess() {
+                    loadNative(context, serverExtras, customEventNativeListener)
+                }
+
+                override fun onFailure(error: Throwable) {
+                    val msg = "InMobi interstitial request failed due to InMobi initialization failed with an exception."
+                    onError(customEventNativeListener, msg)
+                }
+            })
+    }
+
+    private fun onError(customEventNativeListener: CustomEventNativeListener, errorMsg: String) {
+        customEventNativeListener.onNativeAdFailed(NativeErrorCode.NATIVE_ADAPTER_CONFIGURATION_ERROR)
+        MoPubLog.log(
+            getAdNetworkId(),
+            AdapterLogEvent.LOAD_FAILED,
+            TAG,
+            NativeErrorCode.NATIVE_ADAPTER_CONFIGURATION_ERROR,
+            errorMsg
+        )
+    }
+
+    private fun getAdNetworkId(): String {
+        return mPlacementId.toString() ?: ""
+    }
+
+    private fun loadNative(context: Context, serverExtras: Map<String, String>, customEventNativeListener: CustomEventNativeListener) {
+        inMobiStaticNativeAd = try {
+            if (context == null) {
+                MoPubLog.log(AdapterLogEvent.CUSTOM, TAG, "Context passed to the Adapter is null or might have garbage collected")
+                customEventNativeListener.onNativeAdFailed(NativeErrorCode.UNSPECIFIED)
+                return
+            }
+            InMobiNativeAd(context, customEventNativeListener, mPlacementId)
+        } catch (e: SdkNotInitializedException) {
+            MoPubLog.log(AdapterLogEvent.CUSTOM, TAG, "Error while creating InMobiNative Object. " + e.message)
+            handleNativeInitializationFailure()
+            return
+        }
+        inMobiStaticNativeAd?.setExtras(InMobiAdapterConfiguration.inMobiTPExtras)
+
+        val adMarkup = serverExtras[DataKeys.ADM_KEY]
+        if (adMarkup != null) {
+            MoPubLog.log(
+                AdapterLogEvent.CUSTOM, InMobiBanner.ADAPTER_NAME,
+                "Ad markup for InMobi banner ad request is present. Will make Advanced Bidding ad request " +
+                        "using markup: " + adMarkup
+            )
+            inMobiStaticNativeAd?.loadAd(adMarkup?.toByteArray())
```

Comment:
```suggestion
            inMobiStaticNativeAd?.loadAd(adMarkup.toByteArray())
```

As we already have a non null check above

Reply:
done